mava/networks.py

```diff
@@ -151,6 +155,42 @@ def __call__(self, obs_embedding: chex.Array, observation: Observation) -> tfd.I
         )
 
 
+class DiscreteActionEpsGreedyMaskedHead(nn.Module):
+    """Discrete Action Head With Epsilon Greedy Selection And Action Masking"""
+
+    action_dim: int
+
+    @nn.compact
+    def __call__(
+        self, obs_embedding: chex.Array, observation: Observation, epsilon: float
+    ) -> Tuple[chex.Array, tfd.Categorical]:
+        """Action selection for distrete action space environments.
+
+        Args:
+            obs_embedding: Observation embedding from network torso.
+            observation: Observation object containing `agents_view`, `action_mask` and
+                `step_count`.
+
+        Returns:
+            q_values: used for double Q-learning selection.
+            eps_greedy_dist: an eps-greedy initialised tfd.Categorical for sampling actions.
+
+        NOTE: We pass both the observation embedding and the observation object to the action head
+        since the observation object contains the action mask and other potentially useful
+        information.
+        """
```

Comment:
Convention is for extra docs to go 1 line below the main explainer, but before args/return
```suggestion
        """Action selection for distrete action space environments.

        NOTE: We pass both the observation embedding and the observation object to the action head
        since the observation object contains the action mask and other potentially useful
        information.
        
        Args:
            obs_embedding: Observation embedding from network torso.
            observation: Observation object containing `agents_view`, `action_mask` and
                `step_count`.

        Returns:
            q_values: used for double Q-learning selection.
            eps_greedy_dist: an eps-greedy initialised tfd.Categorical for sampling actions.
        """
```